docs/infra/understanding-the-setup.md

```diff
@@ -0,0 +1,166 @@
+# Understanding the infra setup
+
+The infrastructure code is modularized and a bit abstracted for greater re-use.
+It can be overwhelming at first to fit how it's all connected in your head. The
+[/infra/README.md](/infra/README.md) doc has a lot of good overview info. This
+doc aims to provide a little better connective tissue between the architectural
+concepts and the concrete things you'll interact with.
+
+If you want deeper background rationale for the current setup:
+
+- [/docs/decisions/infra/2023-05-09-separate-terraform-backend-configs-into-separate-config-files.md](/docs/decisions/infra/2023-05-09-separate-terraform-backend-configs-into-separate-config-files.md)
+- [/docs/decisions/infra/2023-09-07-consolidate-infra-config-from-tfvars-files-into-config-module.md](/docs/decisions/infra/2023-09-07-consolidate-infra-config-from-tfvars-files-into-config-module.md)
+- [/docs/decisions/infra/2023-09-11-separate-app-infrastructure-into-layers.md](/docs/decisions/infra/2023-09-11-separate-app-infrastructure-into-layers.md)
+
+TL;DR for the why: group resources with a similar lifecycle and scope, reduce
+code duplication and drift between environments, help streamline receiving
+updates from upstream
+
+It's important to understand most of the actual resource code [lives in
+re-usable modules under `/infra/modules/`](/docs/infra/module-architecture.md).
+Everything else largely exists to call that re-usable code with the correct
+parameters at the correct time, [in a well structured
+way](/docs/infra/module-dependencies.md). That "everything else" can be broken
+down into three high-level "slices" (some with further internal divisions).
+
+One last bit of exposition, as yet more new terminology has been introduced with
+"slice". If you've read other documentation already you'll have likely seen
+references to infrastructure "layers", but there are some conceptual
+pseudo-layers that, for clarity, that this document calls a "slice". Some slices
+directly correspond to a layer, other slices correspond to multiple layers. In
+summary:
+
+- Slice: Conceptual or organizational grouping, it does not directly correspond
+  to infrastructure resources itself, though may be expressed in the file
+  structure.
+- Layer: Corresponds to root modules and actual resources.
+
+![Visualization of infra code
+layers](https://lucid.app/publicSegments/view/623affad-8b51-4482-86e2-f1a3ad1bd623/image.png)
```

Comment:
Doesn't have to be in this PR, but the text in the diagram is too small I think, i.e. things are too spread out.